hazelcast/src/main/java/com/hazelcast/util/HashUtil.java

```diff
@@ -53,22 +54,20 @@
     private HashUtil() {
     }
 
-    public static int MurmurHash3_x86_32(byte[] data, int offset, int len) {
-        return MurmurHash3_x86_32(data, offset, len, DEFAULT_MURMUR_SEED);
-    }
-
     /**
-     * Returns the MurmurHash3_x86_32 hash.
+     * Returns the MurmurHash3_x86_32 hash of a block inside a byte array.
      */
     public static int MurmurHash3_x86_32(byte[] data, int offset, int len, int seed) {
+
+        // (len & ~(MURMUR32_BLOCK_SIZE - 1)) is the length rounded down to the Murmur32 block size boundary
+        final int tailStart = offset + (len & ~(MURMUR32_BLOCK_SIZE - 1));
+
         int c1 = 0xcc9e2d51;
         int c2 = 0x1b873593;
 
         int h1 = seed;
-        // round down to 4 byte block
-        int roundedEnd = offset + (len & 0xfffffffc);
 
-        for (int i = offset; i < roundedEnd; i += 4) {
+        for (int i = offset; i < tailStart; i += MURMUR32_BLOCK_SIZE) {
             // little endian load order
             int k1 = (data[i] & 0xff)
```

Comment:
we can read integer directly if the underlying platform is little endian and if `offset` is 4 bytes aligned